### ISSUE TITLE
BatchNorm layers not present in ONNX model

### ISSUE THREAD

**Issue by Kunal-Kumar-Sahoo** [ultralytics/ultralytics#20252]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am exporting the YOLO model to make calculate floating point operations and memory accesses of YOLOv8n model. So when I load the model in Python and I print the model, I find BatchNorm layers present after most of the Conv layers. But upon exporting it to ONNX model, I see the BatchNorm layers are missing in the ONNX model. 

How can I preserve the BatchNorm layers in the exported format.

**Comment by Y-T-G** [comment]
The BatchNorm layers get fused Conv layers for faster inference.

**Comment by Kunal-Kumar-Sahoo** [comment]
Yeah I know that, but I want to know if there is a way to prevent that?

**Comment by Y-T-G** [comment]
You can disable `fuse()`

https://github.com/ultralytics/ultralytics/issues/14826#issuecomment-2258727121

**Comment by Kunal-Kumar-Sahoo** [comment]
I tried the above approach but it doesn't seem to work. Here is the code for your reference:

```python
from ultralytics import YOLO
import os

# Load the YOLOv8 model
model_weights = 'yolov8n.pt'
model = YOLO(model_weights)

def fuse(self, *args, **kwargs):
    return self

model.model.fuse = fuse.__get__(model.model, type(model.model))

# Export the model to ONNX format
model.export(format='onnx', imgsz=640, dynamic=False, optimize=False)
!mv yolov8n.onnx {os.path.join(export_path, 'yolov8n_kunal.onnx')}

print(f"Model exported to ONNX format at: {os.path.join(export_path, 'yolov8n_kunal.onnx')}")
```

**Comment by Y-T-G** [comment]
It seems like PyTorch also performs fusing internally during export

https://github.com/pytorch/pytorch/issues/49226

**Comment by Kunal-Kumar-Sahoo** [comment]
Does Ultralytics library use `torch.onnx.export` to export YOLO models to ONNX formats? Because in torch you can set `do_constant_folding=False` and `training=torch.onnx.TrainingMode.TRAINING` to solve that specific issue.

I tried exporting the YOLO model to ONNX file using torch as well, but was getting some concat errors amongst the output layers.

**Comment by Y-T-G** [comment]
You can find the export code in the `exporter.py` file

https://github.com/ultralytics/ultralytics/blob/51918c42d8d158a2795b0249730db6d6a71a2f15/ultralytics/utils/export.py#L35
